### PR TITLE
Fix/comparison result is always the same

### PR DESCRIPTION
**Pull request by codingdave**
works on issues reported by https://lgtm.com/projects/g/darktable-org/darktable/?mode=tree&ruleFocus=2154840804

### CODE REVIEW

File: src/common/points.h [hunk 1]
```diff
@@ -1056,27 +1056,9 @@ void init_by_array(sfmt_state_t *s, uint32_t *init_key, int key_length)
 {
   int i, j, count;
   uint32_t r;
-  int lag;
-  int mid;
+  int lag = 11;
   int size = N * 4;
```
Comment (TurboGit) [inline src/common/points.h]: I don't think this is correct. N is defied as:

#define N (MEXP / 128 + 1)

And MEXP comes from an include definition which may depends on the platform.

size = N * 4 is 155 if MEXP is set to default (see top of this file), and so path "size  >= 60" will be taken.

File: src/iop/demosaic.c [hunk 2]
```diff
@@ -2073,11 +2073,6 @@ static void xtrans_fdc_interpolate(struct dt_iop_module_t *self, float *out, con
             maxval = (maxval < hm[d] ? hm[d] : maxval);
           }
           maxval -= maxval >> 3;
-          for(int d = 0; d < ndir - 4; d++)
```
Comment (TurboGit) [inline src/iop/demosaic.c]: This looks very suspicious to me!

File: src/iop/liquify.c [hunk 3]
```diff
@@ -2804,7 +2805,7 @@ int mouse_moved (struct dt_iop_module_t *module,
                  int which)
 {
   dt_iop_liquify_gui_data_t *g = (dt_iop_liquify_gui_data_t *) module->gui_data;
-  int handled = g->last_hit.elem ? 1 : 0;
+  bool handled = g->last_hit.elem ? true : false;
```
Comment (TurboGit) [inline src/iop/liquify.c]: no bool (C++ things) we use boolean.

**Comment by codingdave** [comment]
Is it ok to correct the commits (rewriting history) and force push? Or will the discussion above get damaged?

**Comment by TurboGit** [comment]
Yes it is correct to rewrite and force push. It is what I would expect from any topic branch.

**Pull request closed without merging**